rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRestInterface.java

```diff
@@ -152,6 +153,19 @@ boolean nodeIsAvailable(@HeaderParam("sessionid") String sessionId, @QueryParam(
     List<RMNodeSourceEvent> getExistingNodeSources(@HeaderParam("sessionid") String sessionId)
             throws NotConnectedException;
 
+    @POST
+    @Path("nodesource")
+    @Produces("application/json")
+    NSState defineNodeSource(@HeaderParam("sessionid") String sessionId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)